src/Type.zig

```diff
@@ -1214,6 +1326,16 @@ pub fn dump(ty: Type, w: anytype) @TypeOf(w).Error!void {
             try w.writeAll("[<expr>]");
             try ty.data.vla.elem.dump(w);
         },
+        .typeof_type, .decayed_typeof_type => {
+            try w.writeAll("typeof(");
+            try ty.data.sub_type.dump(w);
+            try w.writeAll(")");
+        },
+        .typeof_expr, .decayed_typeof_expr => {
+            try w.writeAll("typeof(<expr>: ");
```

Comment:
Someday I'll go through the effort of adding a `token` field to `Node` and then this can be changed to pretty print the expression like clang.